pkg/scheduler/algorithm/priorities/image_locality_test.go

```diff
@@ -65,39 +123,77 @@ func TestImageLocalityPriority(t *testing.T) {
 		Images: []v1.ContainerImage{
 			{
 				Names: []string{
-					"gcr.io/40",
+					"gcr.io/40:" + parsers.DefaultImageTag,
 					"gcr.io/40:v1",
 					"gcr.io/40:v1",
+					"gcr.io/40@" + getImageFakeDigest("40"),
+					"40:v1",
 				},
 				SizeBytes: int64(40 * mb),
 			},
 			{
 				Names: []string{
-					"gcr.io/140",
+					"gcr.io/140:" + parsers.DefaultImageTag,
 					"gcr.io/140:v1",
+					"gcr.io/140@" + getImageFakeDigest("140"),
+					"140:v1",
 				},
 				SizeBytes: int64(140 * mb),
 			},
 			{
 				Names: []string{
-					"gcr.io/2000",
+					"gcr.io/2000:" + parsers.DefaultImageTag,
 				},
```

Comment:
The missing a case seems to be: 
A pod has `image: gcr.io/nginx` in spec, nodeInfo has `image: gcr.io/nginx:latest`, and this priority will work as expected. 

Isn't this case you originally try to fix? or I missed it something?

Reply:
Exactly! But the original cases would have it cover or wouldn't it?

```
	test40250 := v1.PodSpec{
		Containers: []v1.Container{
			{
				Image: "gcr.io/40",
			},
			{
				Image: "gcr.io/250",
			},
		},
	}
```

Reply:
Oh, I missed that part in diff, thanks.